src/include/migraphx/op/scatter_op.hpp

```diff
@@ -67,38 +72,60 @@ struct scatter_op : op_name<Derived>
         // If non-packed, this converts to a packed output while preserving permutation of tensor
         return inputs.front().with_lens(inputs.front().lens());
     }
+   
+    // iterate through items in shape
+    template <class TensorView0, class TensorView1>
+    void scatter_reduce_iterate(TensorView0 indices, TensorView1 output, TensorView1 update) const
+    {
+        auto ind_s = indices.get_shape();
+        auto output_shape = output.get_shape();
+        auto axis_dim_size = output_shape.lens()[axis];
+        shape_for_each(ind_s, [&](const auto& idx) {
+                auto out_idx = idx;
+
+                // tensor_view::() invokes indexing logic of
+                // std::size_t shape::index(std::size_t i) const
+                auto index = indices(idx.begin(), idx.end());
+
+                // this addition doesn't necessarily make index positive if index was out of bounds
+                index         = (index < 0) ? index + axis_dim_size : index;
+                if(skip_out_of_bounds and index < 0)
+                {
+                    return;
+                }
+                else
+                {
+                    assert(index >= 0);
```

Comment:
move this before the if statement and remove the else case

Reply:
Different behavior, we don't want to assert fail for this case when `skip_out_of_bounds` is enabled. The assert as an else is mainly for debugging purposes. If there is a better way to do this let me know

Reply:
I think `assert((index >=0) xor (skip_out_of_bounds == True))` works?